azuredevops.go

```diff
@@ -182,22 +222,37 @@ func runAzureDevOpsHandler(c *gin.Context) {
 	}
 	if err != nil {
 		fmt.Println("Unable to unmarshal projects: ", err)
-		c.JSON(http.StatusBadRequest, fmt.Sprintf("Error: %+v", err))
+		ginutil.WriteProblemError(c, err, problem.Response{
+			Type: "prob/provider/azuredevops/unmarshal-project-error",
+			Title: "Error unmarshalling project.",
+			Status: http.StatusBadRequest,
+			Detail: "Unable to unmarshal projects.",
+		})
 		return
 	}
 
 	for _, project := range projects.Value {
 		buildDefinitionStr, err := getAzureDevOpsBuildDefinition(i, project.Name)
 		if err != nil {
 			fmt.Println("Unable to get build definition: ", err)
-			c.JSON(http.StatusBadRequest, fmt.Sprintf("Error: %+v", err))
+			ginutil.WriteProblemError(c, err, problem.Response{
+				Type: "prob/provider/azuredevops/getting-build-definition-error",
+				Title: "Error getting build definition.",
+				Status: http.StatusBadRequest,
+				Detail: "Unable to get build definition.",
+			})
```

Comment:
Suggest generalizing to something provider-agnostic like `/prob/provider/fetch-build-definition`

Adding `-error` to the type is unnecessary. It's implicit that it's an error or similar as it's a problem definition.